src/controllers/validator/also-known-as.ts

```diff
@@ -0,0 +1,61 @@
+import type { AlsoKnownAs } from '../../types/shared.js';
+import type { IValidationResult, IValidator, Validatable } from './validator.js';
+
+export class AlsoKnownAsValidator implements IValidator {
+	validateEach(alsoKnownAs: AlsoKnownAs): IValidationResult {
+		if (!alsoKnownAs.uri) {
+			return {
+				valid: false,
+				error: "URI field is required for AlsoKnownAs",
+			};
+		}
+
+		if (typeof alsoKnownAs.uri !== 'string') {
+			return {
+				valid: false,
+				error: "URI field in AlsoKnownAs must be a string",
+			};
+		}
+
+		if (!alsoKnownAs.description) {
+			return {
+				valid: false,
+				error: "Description field is required for AlsoKnownAs",
+			};
+		}
+		if (typeof alsoKnownAs.description !== 'string') {
+			return {
+				valid: false,
+				error: "Description field in AlsoKnownAs must be a string",
+			};
+		}
+		return { valid: true };
+	}
+
+	validate(alsoKnownAs: Validatable): IValidationResult {
+		alsoKnownAs = alsoKnownAs as AlsoKnownAs[];
+		if (!Array.isArray(alsoKnownAs)) {
+			return {
+				valid: false,
+				error: 'AlsoKnownAs must be an array.',
+			};
+		}
+
+		if (alsoKnownAs.length === 0) {
+			return {
+				valid: false,
+				error: 'AlsoKnownAs must have at least one entry.',
+			};
+		}
```

Comment:
Unsure about this. Are we saying this should _always_ be populated? This is not strictly true for DIDs, if it's also being used to validate for DIDs. For DID-Linked Resources, this is true **in our default implementation** but is optional. (Think of the Protobuf definitions.)